src/main/scala/org/constellation/storage/CheckpointService.scala

```diff
@@ -202,13 +293,12 @@ class CheckpointService[F[_]: Sync: LiftIO](
       }
     }
 
-  private def syncPending(cb: CheckpointBlock)(implicit dao: DAO): F[Unit] =
-    Sync[F].delay {
-      pendingAcceptance.synchronized {
-        pendingAcceptance.contains(cb.baseHash).flatMap {
-          case false => pendingAcceptance.put(cb.baseHash, cb)
-          case _    => throw PendingAcceptance(cb)
-        }
+  private[storage] def syncPending(storage: Ref[F, Set[String]], baseHash: String)(implicit dao: DAO): F[Unit] =
```

Comment:
`baseHash` -> `cbHash`

Reply:
It's type agnostic you could sync any storage e.g Transactions